modules/missionmail.js

```diff
@@ -0,0 +1,258 @@
+// @ts-check
+const u = require("../utils/utils");
+const c = require("../config/config.json");
+const receive = require("imapflow");
+const interpret = require("mailparser-mit");
+const htmlparse = require("html-to-text");
+const Augur = require("augurbot-ts");
+const pf = new (require('profanity-matcher'));
+const banned = require('../data/banned.json');
+const { ChannelType, ButtonStyle, Message } = require("discord.js");
+const [approveIdPrefix, rejectIdPrefix] = ["approvemissionmail", "rejectmissionmail"];
+const replyRegexes = [
+  // /^on[\s\n\r]+.+wrote:[\s\n\r]*$/im,
+  // /on[\s\n\r]+.+wrote:[\s\n\r]*/im,
+  /^on[^>]*@[^<]*wrote:\n\n>/im,
+  // /^on\s.+wrote:\s*$/im,
+  // /On\s.+wrote:\s*/im,
+  /^>.*$/m,
+  /^-+ original message -+$/m,
+  /^from:.*$/m,
+  /^sent:.*$/m,
+  /^to:.*$/m,
+  /^subject:.*$/m,
+  /^date:.*$/m
+];
+/** @type {receive.ImapFlow | undefined} */
+let receiver;
+/** @type {Map<string, {approve:() => any, reject:() => any}>} */
+const awaitingMods = new Map();
+async function logNPull() {
+  const creds = c.google.mail;
+  try {
+    receiver?.removeAllListeners();
+    receiver?.close();
+    receiver = new receive.ImapFlow({
+      auth: {
+        user: creds.email,
+        pass: creds.gAccountAppPass
+      },
+      host: 'imap.gmail.com',
+      port: 993,
+    });
+    await receiver.connect();
+    await receiver.mailboxOpen("INBOX");
+    sendUnsent();
+    // this shouldn't run too often, but it could theoretically depending on how much the email server spams.
+    receiver.on("exists", sendUnsent);
+  } catch (error) {
+    u.errorHandler(error, `login & pull missionmail`);
+    receiver = undefined;
+  }
+}
+async function sendUnsent() {
+  if (receiver?.usable) {
+    try {
+      let messageIds = (await receiver.search({ unKeyword: 'icarusForwarded' }));// , from: "*@missionary.org" }))
+      messageIds = messageIds.filter(msgId => {
+        return awaitingMods.has(msgId + "") ? undefined : msgId;
```

Comment:
maybe just `return !awaitingMods.has(msgId)`
also use `u.db.sheets.missionaries` as a whitelist for emails